wandb/sdk/wandb_run.py

```diff
@@ -1669,9 +1669,28 @@ def log_artifact(self, artifact_or_path, name=None, type=None, aliases=None):
         if isinstance(aliases, str):
             aliases = [aliases]
         artifact.finalize()
+        self._assert_can_log_artifact(artifact)
         self._backend.interface.publish_artifact(self, artifact, aliases)
         return artifact
 
+    def _assert_can_log_artifact(self, artifact):
```

Comment:
We need to conditionally call the public API based on whether or not we're in offline mode (which you can get from `_settings._mode` I believe).

If the user is in dryrun / offline mode, we can just skip validation. We have no choice but to error at sync time in that scenario.

Reply:
Great, I added this condition in next commit - thanks for pointing this out